dataland-frontend/src/components/forms/parts/UploadReports.vue

```diff
@@ -0,0 +1,288 @@
+          <div class="form-field">
+            <YesNoComponent
+              :displayName="euTaxonomyKpiNameMappings?.groupLevelIntegratedReport ?? 'Group Level Integrated Report'"
+              :info="euTaxonomyKpiInfoMappings?.groupLevelIntegratedReport ?? 'Group Level Integrated Report'"
+              :name="'isGroupLevel'"
+            />
+          </div>
+        </FormKit>
+      </div>
+    </div>
+  </FormKit>
+</template>
+
+<script lang="ts">
+import { defineComponent } from "vue";
+
+import Calendar from "primevue/calendar";
+import UploadFormHeader from "@/components/forms/parts/UploadFormHeader.vue";
+import PrimeButton from "primevue/button";
+import FileUpload, { FileUploadEmits } from "primevue/fileupload";
+import YesNoComponent from "@/components/forms/parts/YesNoComponent.vue";
+import { formatBytesUserFriendly } from "@/utils/NumberConversionUtils";
+import { WhichSetOfFiles } from "@/components/forms/Types";
+
+export default defineComponent({
+  name: "UploadReports",
+  components: {
+    Calendar,
+    UploadFormHeader,
+    PrimeButton,
+    FileUpload,
+    YesNoComponent,
+  },
+  emits: ["selectedFiles", "removeReportFromFilesToUpload", "updateReportDateHandler"],
+  data() {
+    return {
+      formsDatesFilesToUpload: [] as string[] | undefined,
+      formatBytesUserFriendly,
+    };
+  },
+  methods: {
+    /**
+     * Function to emit event when files are selected
+     *
+     * @param event date in date format
+     * @param event.originalEvent event
+     * @param event.files files
+     */
+    onSelectedFiles(event: { files: Record<string, string>[]; originalEvent: Event }) {
+      this.$emit("selectedFiles", event);
+    },
+    /**
+     * Function to emit event when files are selected remove report from files to upload
+     *
+     * @param fileToRemove File To Remove
+     * @param fileRemoveCallback Callback function removes report from the ones selected in formKit
+     * @param index Index number of the report
+     */
+    removeReportFromFilesToUpload(
+      fileToRemove: Record<string, string>,
+      fileRemoveCallback: (x: number) => void,
+      index: number
+    ) {
+      this.$emit("removeReportFromFilesToUpload", fileToRemove, fileRemoveCallback, index);
+    },
+    /**
+     * Function to emit event to update the date of a single report file
+     *
+     * @param index file to update
+     * @param event new date value
+     * @param whichSetOfFiles which set of files will be edited
+     */
+    updateReportDateHandler(index: number, event: Date, whichSetOfFiles: WhichSetOfFiles) {
+      this.$emit("updateReportDateHandler", index, event, whichSetOfFiles);
+    },
+    /**
+     * Function to clear all not uploaded files
+     *
+     */
+    clearAllNotUploadedFiles(): void {
+      (this.$refs.fileUpload as FileUploadEmits).clear();
+    },
+  },
+
+  props: {
+    filesToUpload: {
+      type: Array,
+    },
+    uploadFiles: {
```

Comment:
I strongly suggest to rename this prop. I have made some suggestions for a different name in one of my other comments.
But also feel free to pick something different if you have a good idea.

The way I understand this prop is that it contains elements of type `ExtendedCompanyReport` which represent reports that have already been uploaded for an existing dataset. But those elements do not contain the actual file!
They just contain a reference (documentId as reference string, right?).

Reply:
I changed the variable names to:
uploadFiles => listOfUploadedReportsInfo
namesOfFilesToUpload => namesOfAllCompanyReportsForTheDataset

2) Yes, thats right